java/src/jmri/jmrit/display/Editor.java

```diff
@@ -650,9 +652,17 @@ public void paint(Graphics g) {
                 g2d.scale(_paintScale, _paintScale);
             }
 
-            super.paint(g);
-            paintTargetPanel(g);
-
+            try {
+               super.paint(g);
+               paintTargetPanel(g);
+            } catch (Exception e) {
+                log.error("paint failed in thread "+
+                    Thread.currentThread().getName()+" "+Thread.currentThread().getId()+": "+e);
+                StringWriter outError = new StringWriter();
+                e.printStackTrace(new PrintWriter(outError));
+                log.error(outError.toString());
```

Comment:
Instead of manually getting a stack trace and logging it, it's better to just include the Exception in the original log request:
```
log.error("paint failed in thread "+Thread.currentThread().getName()+" "+Thread.currentThread().getId(), e);
```

(Note comma, not plus sign).  That extra argument means "Include this exception in the log".

It also works with the new-format way of making messages:
```
log.error("paint failed in thread {} : {}", Thread.currentThread().getName(), Thread.currentThread().getId(), e);
```

The thread is automatically logged, and printed (when configured) in the log file output, so the simplest form that logs what you want is:

```
log.error("paint failed", e);
```